### PR TITLE
GitHub integration

### PR DESCRIPTION
**Pull request by serhii-lekariev**
This PR adds the implementation of the part of `publishing` that deals with Git and GitHub.

One of the goals of the application is to keep our repositories in sync with the freshest library versions.
In order to do this, we need to automate the creation of version-bumping commits.

#### Git

In order to change the remote, one needs to push a branch. `io.spine.publishing.git` provides a basic Git API for creating branches, committing changes, and pushing them to the remote repos.

The changes are pushed directly to `master`.

### CODE REVIEW

**Comment by serhii-lekariev** [comment]
@armiol, PTAL.

**Comment by serhii-lekariev** [comment]
@armiol, PTAL again.

**Comment by serhii-lekariev** [comment]
@armiol, PTAL again.

File: src/main/kotlin/io/spine/publishing/github/VersionUpdate.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * changed to have the correct version;
+ * 3) the commit is performed;
+ * 4) the local `master` branch is pushed to the respective remote repository.
+ */
+class VersionUpdate(private val remote: RemoteRepository,
+                    private val credentials: CredentialsProvider) {
+
+    /**
+     * Returns the list of commands to execute in order to push a version bump branch to the remote
+     * repo.
```
Comment (armiol) [inline src/main/kotlin/io/spine/publishing/github/VersionUpdate.kt:42]: Let's not leave orphaned words. Always make sure to move more text to the new line in such cases.

File: src/main/kotlin/io/spine/publishing/gradle/InterdependentLibraries.kt [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.publishing.gradle
+
+/**
+ * A number of libraries that have dependencies between them.
+ *
+ * Allows to [update and publish new versions][InterdependentLibraries.publish] in a
+ * dependency-based order.
```
Comment (armiol) [inline src/main/kotlin/io/spine/publishing/gradle/InterdependentLibraries.kt:27]: "Dependency-based" order explains nothing, really. It still leaves some questions. Please describe.

File: src/main/kotlin/io/spine/publishing/gradle/InterdependentLibraries.kt [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+class InterdependentLibraries(private val ordering: Ordering) {
+
+    /**
+     * Goes through the libraries one by one, updating their version, building and publishing
+     * new versions them to the remote artifact repositories.
+     *
+     * First, updates the versions of all libraries. Then, starts building and publishing the
+     * libraries in a dependency-based order, e.g:
+     *
+     * library A depends on library B and C. Library B depends on library C.
```
Comment (armiol) [inline src/main/kotlin/io/spine/publishing/gradle/InterdependentLibraries.kt:38]: I would draw a simple picture with pseudo-graphics explaining the dependencies.

File: src/main/kotlin/io/spine/publishing/gradle/InterdependentLibraries.kt [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+     * In such a configuration, the following happens:
+     *
+     * 1) library C is built;
+     * 2) library C is published to the local Maven repo, so that it can be used for further builds;
+     * 3) library B is built;
+     * 4) library B is published to the local Maven repo;
+     * 5) library A is built;
+     * 6) library A is published to the local Maven repo;
+     * 7) library A, B and C are published to the remote artifact repository, as at this point
```
Comment (armiol) [inline src/main/kotlin/io/spine/publishing/gradle/InterdependentLibraries.kt:47]: Aren't you tired of typing the word "library"? Why "A", "B" and "C" are not enough?

**Comment by serhii-lekariev** [comment]
@armiol, PTAL.

**Comment by serhii-lekariev** [comment]
@armiol, PTAL.

**Comment by serhii-lekariev** [comment]
@armiol, PTAL again.

**Pull request merged**